node/pkg/watchers/aptos/watcher.go

```diff
@@ -100,17 +121,181 @@ func (e *Watcher) Run(ctx context.Context) error {
+				version := event.Get("version")
+				sequence := event.Get("sequence_number")
+				if !version.Exists() || !sequence.Exists() {
+					continue
+				}
+
+				versionNum := version.Uint()
+				sequenceNum := sequence.Uint()
+
+				if err := e.processTransactionVersion(logger, versionNum, sequenceNum, true); err != nil {
+					logger.Error("Failed to process transaction for reobservation",
+						zap.Uint64("version", versionNum),
+						zap.Uint64("sequence", sequenceNum),
+						zap.Error(err))
+					continue
+				}
+			}
+
+		case <-timer.C:
+			eventsQuery := ""
+
+			if nextSequence == 0 {
+				// if nextSequence is 0, we look up the most recent event
+				eventsQuery = fmt.Sprintf(`%s/v1/accounts/%s/events/%s?limit=1`, e.aptosRPC, e.aptosAccount, e.aptosHandle)
+			} else {
+				// otherwise just look up events starting at nextSequence.
+				eventsQuery = fmt.Sprintf(`%s/v1/accounts/%s/events/%s?start=%d`, e.aptosRPC, e.aptosAccount, e.aptosHandle, nextSequence+1)
+			}
+
+			eventsJson, err := e.retrievePayload(eventsQuery)
+			if err != nil {
+				logger.Error("retrievePayload", zap.Error(err))
+				p2p.DefaultRegistry.AddErrorCount(e.chainID, 1)
+				continue
+			}
+
+			// data doesn't exist yet. skip, and try again later
+			if string(eventsJson) == "" {
+				continue
+			}
+
+			if !gjson.Valid(string(eventsJson)) {
+				logger.Error("InvalidJson: " + string(eventsJson))
+				p2p.DefaultRegistry.AddErrorCount(e.chainID, 1)
+				continue
+			}
+
+			events := gjson.ParseBytes(eventsJson)
+			for _, event := range events.Array() {
+				version := event.Get("version")
+				sequence := event.Get("sequence_number")
+				if !version.Exists() || !sequence.Exists() {
+					continue
+				}
+
+				versionNum := version.Uint()
+				sequenceNum := sequence.Uint()
+
+				logger.Debug("Found event from REST API",
+					zap.Uint64("version", versionNum),
+					zap.Uint64("sequence_number", sequenceNum))
+
+				if err := e.processTransactionVersion(logger, versionNum, sequenceNum, false); err != nil {
+					logger.Error("Failed to process transaction",
+						zap.Uint64("version", versionNum),
+						zap.Uint64("sequence", sequenceNum),
+						zap.Error(err))
+					continue
+				}
+
+				// Update nextSequence to track progress
+				if sequenceNum > nextSequence {
+					nextSequence = sequenceNum
+				}
+			}
+
+			// Health check
+			aptosHealth := fmt.Sprintf(`%s/v1`, e.aptosRPC)
+			health, err := e.retrievePayload(aptosHealth)
+			if err != nil {
+				logger.Error("health", zap.Error(err))
+				p2p.DefaultRegistry.AddErrorCount(e.chainID, 1)
+				continue
+			}
+
+			if !gjson.Valid(string(health)) {
+				logger.Error("Invalid JSON in health response: " + string(health))
+				p2p.DefaultRegistry.AddErrorCount(e.chainID, 1)
+				continue
+			}
+
+			logger.Debug(string(health) + string(eventsJson))
+
+			pHealth := gjson.ParseBytes(health)
+			blockHeight := pHealth.Get("block_height")
+
+			if blockHeight.Uint() > math.MaxInt64 {
```

Comment:
This type check looks wrong? (or maybe the log message should be more clear). Is the idea that the max block height for Aptos is MaxInt64?

Reply:
Hmm OK the limitation seems to come from the Protobuf for Heartbeat messages: https://github.com/wormhole-foundation/wormhole/blob/565a8d1aff42fb5087dc6a6581b4dc01a8f1d9c2/node/pkg/proto/gossip/v1/gossip.pb.go#L1104-L1121

It's a little funky but not in-scope for this PR.